client/client-multi/client-c/test/test_kaa_channel_manager.c

```diff
@@ -458,7 +464,24 @@ void test_get_bootstrap_client_sync_serialize(void **state)
     kaa_channel_manager_destroy(channel_manager);
 }
 
+void test_process_auth_failure(void **state)
+{
+    (void)state;
 
+    kaa_error_t error_code;
+    kaa_channel_manager_t *channel_manager = NULL;
+    kaa_auth_failure_reason reason;
+
+    error_code = kaa_channel_manager_create(&channel_manager, &kaa_context);
+    ASSERT_EQUAL(error_code, KAA_ERR_NONE);
+    kaa_channel_manager_set_auth_failure_handler(channel_manager, test_auth_failure_handler, &reason);
+
+    kaa_channel_manager_process_auth_failure(channel_manager, KAA_AUTH_STATUS_BAD_CREDENTIALS);
+    ASSERT_EQUAL(reason, KAA_AUTH_STATUS_BAD_CREDENTIALS);
+
+    kaa_channel_manager_process_auth_failure(channel_manager, KAA_AUTH_STATUS_VERIFICATION_FAILED);
+    ASSERT_EQUAL(reason, KAA_AUTH_STATUS_VERIFICATION_FAILED);
```

Comment:
Don't use the macro. Have you read our [testing guide](https://github.com/kaaproject/kaa/blob/develop/client/client-multi/client-c/docs/testing.org)?